### PR TITLE
feat(components): Add DataList.StatusBar

### PR DESCRIPTION
**Pull request by rodrigoeidelvein**
## Motivations

Adds the ability to add a `DataList.StatusBar` between the filters and the header.

image

### Added

- Added `DataList.StatusBar`

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

You can checkout commit [b6c1a8f](https://github.com/GetJobber/atlantis/pull/1596/commits/b6c1a8f0b593c8bb17d71766ca47749ebda8fb14) or add the snipper below in DataList

```tsx
<DataList.StatusBar>
  <Banner type="error" icon="alert" dismissible>
    Something went wrong. Refresh or check your internet connection.
  </Banner>
</DataList.StatusBar>
```

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

JOB-81344

### CODE REVIEW

File: docs/components/DataList/DataList.stories.mdx [hunk 1]
```diff
@@ -225,3 +225,20 @@ need to set the `filtered` to `true`.
   />
 </DataList>
 ```
+
+### Banner
+
+You can add a banner that will be displayed between the filters and the header.
+This banner can be used to display error messages or anything you want to
+information the user about something related to the Datalist. It renders a
+Atlantis
```
Comment (darryltec) [inline docs/components/DataList/DataList.stories.mdx:234]:
```suggestion
inform the user about something related to the Datalist. It renders an
Atlantis
```

File: docs/components/DataList/DataList.stories.mdx [hunk 2]
```diff
@@ -225,3 +225,20 @@ need to set the `filtered` to `true`.
...
+[Banner](../?path=/docs/components-status-and-feedback-banner-docs--page)
+component
```
Comment (darryltec) [inline docs/components/DataList/DataList.stories.mdx:236]:
```suggestion
component.
```

**Comment by chris-at-jobber** [comment]
I'm not sure I'm sold on the naming of this subcomponent as "DataList Banner". It feels 
a bit too declarative (if we decided to present errors in a different format than Banner, this breaks), and the documentation is basically just describing the Banner component.

If we want to be more specific about _current_ usage, something like DataList Error State would be more helpful, and if we want to be a bit more general about other possible usages, something like DataList Status (DataList StatusMessage?) could be more flexible.

**Comment by rodrigoeidelvein** [comment]
@chris-at-jobber Yeah, I didn't want to keep it just for errors because it might be useful for other usages like giving tips, warnings, and notices... it seems to me that it could be useful other than just being an error banner.

`BannerMessage`, maybe? 🤔

**Comment by chris-at-jobber** [comment]
I'd leave Banner out of it as that's really coupled to the specific current implementation. If we decided to use a StatusLabel or maybe just some red text to indicate the same information, the name would be really confusing, where if we just leave it as "DataList Message or "DataList Status Message" it's more flexible.

**Comment by rodrigoeidelvein** [comment]
@chris-at-jobber okay, I have an idea. What about calling it "StatusBar" (props to Darryl for this name), and then we reserve that space to "status" things, it could be a Banner, a StatusLabel, or however the consumer wants to provide a status there and render its children.

Something like this:
```tsx
<DataList.StatusBar>
  <Banner type="error">message here</Banner>
</DataList.StatusBar>
```

**Comment by chris-at-jobber** [comment]
Nice! I like that

File: packages/components/src/DataList/DataList.tsx [hunk 3]
```diff
@@ -245,3 +251,8 @@ DataList.BatchAction = function DataListBatchAction(
 ) {
   return <DataListAction {...props} />;
 };
+
+/**
+ * Defines a banner that is going to rendered between the filters and the header.
```
Comment (darryltec) [inline packages/components/src/DataList/DataList.tsx:256]:
```suggestion
 * Defines a banner that is going to be rendered between the filters and the header.
```

File: packages/components/src/DataList/components/DataListStatusBar/DataListStatusBar.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  const component = getCompoundComponent<DataListStatusBarProps>(
+    parentChildren,
+    DataListStatusBar,
+  );
+
+  if (!component) return null;
+
+  const children = component?.props.children;
+
+  return <>{children && children}</>;
```
Comment (darryltec) [inline packages/components/src/DataList/components/DataListStatusBar/DataListStatusBar.tsx:30]:
```suggestion
  const children = component.props.children;

  return <>{children}</>;
```

should suffice

**Review by darryltec — CHANGES_REQUESTED** [review]
just a few little things

File: packages/components/src/DataList/DataList.tsx (hunk 3 above)
Comment (chris-at-jobber) [reply packages/components/src/DataList/DataList.tsx:256]:
```suggestion
 * Defines a banner that is rendered between the filters and the header.
```

Even shorter

Comment (darryltec) [reply packages/components/src/DataList/DataList.tsx:256]: even better

File: docs/components/DataList/DataList.stories.mdx [hunk 5]
```diff
@@ -225,3 +225,19 @@ need to set the `filtered` to `true`.
   />
 </DataList>
 ```
+
+### Status Bar
```
Comment (chris-at-jobber) [inline docs/components/DataList/DataList.stories.mdx:229]:
```suggestion
### Status bar
```

Sentence case per our[ guidelines](https://job-80005-add-datalist-banne.atlantis.pages.dev/?path=/docs/content-formatting--page#capitalization)

**Comment by chris-at-jobber** [comment]
The screenshot looks good and I'm happy with the naming - I don't want to block the PR on this but it would be great to have a story added, as since it's not a prop it's hard to see otherwise.

File: packages/components/src/Banner/Banner.tsx
[the remark this answers is not in the document]
Comment (rodrigoeidelvein) [reply packages/components/src/Banner/Banner.tsx:1]: Nope! Reverting it

File: packages/components/src/Banner/index.ts [hunk 6]
```diff
@@ -1 +1 @@
-export { Banner, BannerType } from "./Banner";
+export { Banner, BannerType, BannerProps } from "./Banner";
```
Comment (darryltec) [inline packages/components/src/Banner/index.ts:1]:
```suggestion
export { Banner, BannerType } from "./Banner";
```

this is the error

Comment (rodrigoeidelvein) [reply packages/components/src/Banner/index.ts]: Yeah, I was just pushing a commit 🤦

**Review by darryltec — APPROVED** [review]
ship it!

**Pull request merged**